Sources/StreamChatUI/Deprecations.swift

```diff
@@ -165,3 +165,56 @@ extension Components {
         }
     }
 }
+
+// MARK: - Deprecation of ChatMessageLayoutOptions as an OptionSet
+
+/// Previously `ChatMessageLayoutOptions` was an `OptionSet`, this limited the customization on
+/// the customer side because the raw value needs to be an `Int`. A more flexible approach is to just
+/// have a `Set` of `ChatMessageLayoutOption`. So for backwards compatibility we created the following
+/// typealias `typealias = Set<ChatMessageLayoutOption>` and provided an API like the `OptionSet` so we
+/// don't break the public API.
+
+extension ChatMessageLayoutOptions {
+    @available(*, deprecated, message: "use `id` instead.")
+    var rawValue: String {
+        id
+    }
+
+    @available(*, deprecated, message: "use `remove(_ member: ChatMessageLayoutOption)` instead.")
+    public mutating func remove(_ options: ChatMessageLayoutOptions) {
+        self = subtracting(options)
+    }
+
+    @available(*, deprecated, message: "use `insert(_ member: ChatMessageLayoutOption)` instead.")
+    public mutating func insert(_ options: ChatMessageLayoutOptions) {
+        options.forEach { self.insert($0) }
+    }
+
+    @available(*, deprecated, message: "use `subtracting(_ other: Sequence)` instead.")
+    public mutating func subtracting(_ option: ChatMessageLayoutOption) {
+        self = subtracting([option])
+    }
+
+    @available(*, deprecated, message: "use `intersection(_ other: Sequence)` instead.")
+    public mutating func intersection(_ option: ChatMessageLayoutOption) {
+        self = intersection([option])
+    }
+
+    @available(*, deprecated, message: """
+        use `contains(_ member: ChatMessageLayoutOption` instead. And make sure the custom option is being extended in
+        `ChatMessageLayoutOption` and not in `ChatMessageLayoutOptions`.
+    """)
+    public func contains(_ options: ChatMessageLayoutOptions) -> Bool {
+        let intersected = intersection(options)
+        return intersected.count == options.count
```

Comment:
let set1 = Set(arrayLiteral: 1,2,3)
let set2 = Set(arrayLiteral: 2,3)

set2.isSubset(of: set1) // true
set1.isSuperset(of: set2) // true

Reply:
This is a deprecated method, which most likely nobody is using or will ever use, so I don't mind the current implementation even the one you provided is more correct. But I can change it if you think it is valuable

Reply:
I've updated it anyways 👍